src/openms/source/ANALYSIS/ID/FragmentIndex.cpp

```diff
@@ -252,6 +273,147 @@ namespace OpenMS
       return hits;
   }
 
+  void FragmentIndex::queryPeak(OpenMS::FragmentIndex::SpectrumMatchesTopN& candidates,
+                                const OpenMS::Peak1D& peak,
+                                std::pair<size_t, size_t> candidates_range,
+                                int16_t isotope_error,
+                                uint16_t precursor_charge)
+  {
+      for (uint16_t fragment_charge = 1; fragment_charge <= max_fragment_charge_; fragment_charge++)
+      {
+        for (auto hit : query(peak, candidates_range, fragment_charge))
```

Comment:
might make unnecessary copies of hit.
https://stackoverflow.com/questions/30448182/is-it-safe-to-use-a-c11-range-based-for-loop-with-an-rvalue-range-init